### PR TITLE
RYA-306 Remove rya.benchmark committed gen

### PR DESCRIPTION
**Pull request by ejwhite922**
## Description
Deleted the committed generated source code in rya.benchmark from git.  It is now generated and ignored by git so when building changes to those files they don't show up as changed files in git.
Also, other projects that use generated source now append the license header to the generated files.  The license header is placed in the top level Rya project under "resources" so all projects can access it and has a property in the rya parent pom called "license.header.file".

### Tests
Build

### Links
[Jira](https://issues.apache.org/jira/browse/RYA-306)

### Checklist
- [ ] Code Review
- [ ] Squash Commits

#### People To Review
@meiercaleb 
@amihalik 
@jdasch

### CODE REVIEW

File: extras/rya.benchmark/pom.xml [hunk 1]
```diff
@@ -106,11 +106,29 @@
                 </executions>
                 <configuration>
                     <!-- Place the generated source within the 'src' directory so license-maven-plugin will find it. -->
-                    <outputDirectory>src/main/gen</outputDirectory>
+                    <outputDirectory>src/gen/java</outputDirectory>
```
Comment (jdasch) [inline extras/rya.benchmark/pom.xml]: Line 109 is unnecessary.  Remove so code is stored in ${project.build.directory}/generated-sources per maven conversion.

File: extras/rya.benchmark/pom.xml [hunk 2]
```diff
@@ -106,11 +106,29 @@
...
                     <packageName>org.apache.rya.benchmark.query</packageName>
                 </configuration>
             </plugin>
             
+            <plugin>
```
Comment (jdasch) [inline extras/rya.benchmark/pom.xml]: Plugin execution is unnecessary if code is stored in ${project.build.directory}/generated-sources.

File: extras/rya.export/export.accumulo/pom.xml [hunk 3]
```diff
@@ -102,6 +102,24 @@ under the License.
                     <outputDirectory>src/gen/java</outputDirectory>
                 </configuration>
             </plugin>
+
```
Comment (jdasch) [inline extras/rya.export/export.accumulo/pom.xml]: Plugin execution is unnecessary if code is stored in ${project.build.directory}/generated-sources.

File: extras/rya.export/export.api/pom.xml [hunk 4]
```diff
@@ -98,6 +98,24 @@ under the License.
                 </configuration>
             </plugin>
 
+            <plugin>
```
Comment (jdasch) [inline extras/rya.export/export.api/pom.xml]: Plugin execution is unnecessary if code is stored in ${project.build.directory}/generated-sources.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/332/

**Comment by jdasch** [comment]
Sorry for so many comments.  The main concern I have with this is that if generated source is not stored in the `${project.build.directory}` (aka target), you really can't do a true `mvn clean` without extra build configuration.  So following the maven convention gives you a lot for free and makes your poms small.

**Comment by jdasch** [comment]
I feel bad taking up your time with this Eric.  I've got a [branch](https://github.com/jdasch/incubator-rya/tree/RYA-306) from a while ago that is probably good enough.  Do you want me to update it and send it out as a PR, or let you mod this PR?  Happy to do either.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/357/<h2>Build result: FAILURE</span></h2>[...truncated 5.53 MB...][INFO] Apache Rya Spark Support ........................... SKIPPED[INFO] Apache Rya Web Projects ............................ SKIPPED[INFO] Apache Rya Web Implementation ...................... SKIPPED[INFO] ------------------------------------------------------------------------[INFO] BUILD FAILURE[INFO] ------------------------------------------------------------------------[INFO] Total time: 24:17 min[INFO] Finished at: 2017-08-04T19:23:25+00:00[INFO] Final Memory: 419M/3343M[INFO] ------------------------------------------------------------------------[ERROR] Failed to execute goal com.mycila:license-maven-plugin:2.6:format (default) on project rya.export.api: Resource ${rya.project.basedir}/resources/LICENSE_HEADER.txt not found in file system, classpath or URL: no protocol: ${rya.project.basedir}/resources/LICENSE_HEADER.txt -> [Help 1][ERROR] [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.[ERROR] Re-run Maven using the -X switch to enable full debug logging.[ERROR] [ERROR] For more information about the errors and possible solutions, please read the following articles:[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException[ERROR] [ERROR] After correcting the problems, you can resume the build with the command[ERROR]   mvn <goals> -rf :rya.export.apichannel stoppedSetting status of 69dc245a1fd7b6be21d2309458bf1c7042019dde to FAILURE with url https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/357/ and message: 'FAILURE 'Using context: Jenkins: clean package -Pgeoindexing

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/359/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/364/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/363/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/362/

**Comment by ejwhite922** [comment]
I made all changes requested.  Let me know if anything else is needed.

**Comment by jdasch** [comment]
Looks good.  I just made a pull request to your branch with my last couple nits.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-rya-master-with-optionals-pull-requests/366/

**Pull request closed without merging**